New page: aria-disabled (#10163)
mdn/content: files/en-us/web/accessibility/aria/attributes/aria-disabled/index.md

---
title: 'aria-disabled'
slug: Web/Accessibility/ARIA/Attributes/aria-disabled
tags: 
  - Accessibility
  - ARIA
  - ARIA attribute
  - ARIA property
  - aria-disabled
  - Reference
---

The `aria-disabled` state indicates that the element is perceivable but disabled, so it is not editable or otherwise operable.

## Description

> You generally want users to fill out all the required fields before submitting a form. One way to ensure all required fields are complete is to disable the submit button; but you still want that submit button to be perceivable to ALL users. That is where the `aria-disabled` attribute comes in.

Reviewer: ```suggestion
You generally want users to fill out all the required fields before submitting a form. One way to ensure all required fields are complete is to disable the submit button. However, you still want that submit button to be perceivable to <strong>all</strong> users. That is where the `aria-disabled` attribute comes in.
```

Trying to lower the reading level here a tad, as well as avoiding hardcoded all caps.

> The `aria-disabled` attribute, when set to `true`, indicates to assistive technologies that the element upon which it is set is disabled. It tells your user the element is not editable or otherwise operable without making the element imperceivable.

Reviewer: ```suggestion
The `aria-disabled` attribute, when set to `true`, indicates to assistive technologies that the element upon which it is set is disabled. This declaration tells your user the element is not editable or otherwise operable without making the element imperceivable.
```

Clarifying what "it" is.

> There are several elements you may want to disable, but still want the user to know they exist. Some examples include irrelevant options in a radio group, the header button element associated with non-collapsible accordion panel, a button when the action associated with the button is unavailable, the icon for the currently visible slide in a slideshow slide picker, and items in a menu that are not currently selectable. In each of these cases, you want the user to know the element is there even though the functionality of that control is removed or "disabled".

Reviewer: ```suggestion
There are several elements you may want to disable, but still want the user to know they exist. Some examples include:

- Irrelevant options in a radio group, 
- The header button element associated with non-collapsible accordion panel, 
- A button when the action associated with the button is unavailable, 
- The icon for the currently visible slide in a slideshow slide picker, and 
- Items in a menu that are not currently selectable. 

In each of these cases, you want the user to know the element is there even though the functionality of that control is removed or "disabled".
```

Breaking this out into a list to lower the reading level a bit. Good examples, though!

ARIA attributes only provide semantics. To semantically disable an element without removing it from focus order or from the accessibility tree, set `aria-disabled="true"` on the element. You should include JavaScript to disable the functionality of the element while also changing the appearance of the element so sighted users know it is disabled.

> Note that the state of being disabled applies to the element with `aria-disabled` and all of its focusable descendants. 

Reviewer: Should we use a note component for this?

Author: done! added a brief note above then did a shorter pull out note here.

The `disabled` Boolean attribute provides CSS styles and semantics and removes the ability to click or focus while not disabling hover. By removing the ability to focus and removing it from the accessibility tree, it makes it invisible to assistive technology users. For good user experience, you want to make sure everyone can access all the visible content, no matter how they access the web. It is important to be aware that using the `disabled` attribute can harm usability. 

While adding `disabled` to an HTML form control causes `:disabled` user-agent styles to be applied, adding `aria-disabled="true"` doesn't, by default, alter an element's appearance. The element can be styled with the [attribute selector](/en-US/docs/Web/CSS/Attribute_selectors) `[aria-disabled="true"]

```css
[aria-disabled="true"] {
  opacity: 0.5;
}
```

> > **Note:** Do not use CSS's [`pointer-events: none;`](/en-US/docs/Web/CSS/pointer-events) to make an element non clickable. While it prevents mouse clicks, the element will still be 'clickable' via the keyboard.

Reviewer: ```suggestion
> **Note:** If you are using CSS's [`pointer-events: none;`](/en-US/docs/Web/CSS/pointer-events) to make an element non-clickable, make sure you disable interactivity with JavaScript as well. `pointer-events: none;` prevents mouse clicks, but does not prevent the element from being activated via the keyboard.
```

I rephrased this a bit so it couldn't be construed as avoiding ``pointer-events: none;`` all together. I think use of this declaration can be helpful, but to your larger point, should be part of a larger, more comprehensive solution.

```js
function toggleDisabled(element, status, update) {
  if(status) {
    //element.input.disabled = false;
    element.setAttribute('aria-disabled', 'false');
    update.textContent = 'The element is now enabled.';
    element.removeEventListener('click', function (event) {
      event.preventDefault();
    }
  } else {
    //element.input.disabled = true;
    element.setAttribute('aria-disabled', 'true');
    update.textContent = 'The element is now disabled.';
    element.addEventListener('click', function (event) {
      event.preventDefault();
    }
  }
}
```

> When toggling from `aria-disabled="true"` to `"false"`, use JavaScript to 1) toggle the value to `false`, 2) enable the element, and 3) let the user know the control is now enabled. If you used CSS to style   the disabled state using an attribute selector, the selector will no longer match and the disabled styling will no longer be in effect.

Reviewer: ```suggestion
When toggling from `aria-disabled="true"` to `"false"`, use JavaScript to: 

1. Toggle the value to `false`, 
2. Enable the element, and 
3. Let the user know the control is now enabled. 

If you used just CSS to style the disabled state using an attribute selector, the selector will no longer match and the disabled styling will no longer be in effect.
```
